src/algorithms/base_algorithm.cc

```diff
@@ -123,3 +126,18 @@ bayesmix::MarginalState BaseAlgorithm::get_state_as_proto(unsigned int iter) {
 
   return iter_out;
 }
+
+void BaseAlgorithm::read_params_from_proto(
+    const bayesmix::AlgorithmParams &params) {
+  // Generic parameters
+  maxiter = params.iterations();
+  burnin = params.burnin();
+  init_num_clusters = params.init_num_clusters();
+  auto &rng = bayesmix::Rng::Instance().get();
+  rng.seed(params.rng_seed());
+  // Algorithm-specific parameters
+  if (get_id() == bayesmix::AlgorithmId::Neal8) {
```

Comment:
this could be avoided by having `read_params_from_proto` as virtual and in Neal8 overriding it and calling

```
void Neal8::read_params_from_proto(xxx) {
  Base::read_params_from_proto(xxx);
  set_n_aux(xxx.neal8_n_aux());
}
```

I think it makes stuff easier